src/Mod/Fem/App/FemVTKTools.cpp

```diff
@@ -777,6 +780,16 @@ void FemVTKTools::exportFreeCADResult(const App::DocumentObject* result, vtkSmar
     const Fem::FemResultObject* res = static_cast<const Fem::FemResultObject*>(result);
     const vtkIdType nPoints = grid->GetNumberOfPoints();
 
+    // we need the coresponding mesh to get the correct id for the result data (when the freecad smesh mesh has gaps in the points
+    // vtk has more point. Vtk does not support point gaps, thus the gaps are filled with points. Than the mapping must be correct)
```

Comment:
2 corrections:
`vtk has more _points_.` ... `_Then_ the mapping must be correct)`

Reply:
fixed